d2game/d2player/box.go

```diff
@@ -0,0 +1,529 @@
+package d2player
+
+import (
+	"image/color"
+	"log"
+
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2interface"
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2resource"
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2util"
+	"github.com/OpenDiablo2/OpenDiablo2/d2core/d2asset"
+	"github.com/OpenDiablo2/OpenDiablo2/d2core/d2gui"
+	"github.com/OpenDiablo2/OpenDiablo2/d2core/d2ui"
+)
+
+const (
+	boxSpriteHeight = 15 - 3
+	boxSpriteWidth  = 14 - 2
+)
+
+const (
+	boxCornerTopLeft = iota
+	boxCornerTopRight
+	boxTopHorizontalEdge1
+	boxTopHorizontalEdge2
+	boxTopHorizontalEdge3
+	boxTopHorizontalEdge4
+	boxTopHorizontalEdge5
+	boxTopHorizontalEdge6
+	boxCornerBottomLeft
+	boxCornerBottomRight
+	boxSideEdge1
+	boxSideEdge2
+	boxSideEdge3
+	boxSideEdge4
+	boxSideEdge5
+	boxSideEdge6
+	boxBottomHorizontalEdge1
+	boxBottomHorizontalEdge2
+	boxBottomHorizontalEdge3
+	boxBottomHorizontalEdge4
+	boxBottomHorizontalEdge5
+	boxBottomHorizontalEdge6
+)
+
+type LabelButton struct {
+	label      string
+	callback   func()
+	hoverColor color.RGBA
+	canHover   bool
+	isHovered  bool
+	layout     *d2gui.Layout
+	x, y       int
+}
+
```

Comment:
It seems like `LabelButton` and `Box` really belong in our ui package. Is there a good reason why this stuff is here in `d2player` ?

Reply:
yep, it's gonna be moved un d2gui, I just did everything here while coding the whole thing